imapclient/imapclient.py

```diff
@@ -1721,11 +1721,9 @@ def _send_literal(self, tag, item):
         logger.debug("   (literal) > %s", debug_trunc(item, 256))
         self._imap.send(item)
 
-    def _command_and_check(self, command, *args, **kwargs):
-        unpack = pop_with_default(kwargs, "unpack", False)
-        uid = pop_with_default(kwargs, "uid", False)
-        assert not kwargs, "unexpected keyword args: " + ", ".join(kwargs)
```

Comment:
I think at some point kwargs was passed down to other calls but not any more so what you've done makes much more sense.